system/Validation/StrictRules/FormatRules.php

```diff
@@ -0,0 +1,424 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Validation\StrictRules;
+
+use CodeIgniter\Validation\FormatRules as NonStrictFormatRules;
+
+/**
+ * Format validation Rules.
+ */
+class FormatRules
+{
+    /**
+     * @var NonStrictFormatRules
+     */
+    private $nonStrictFormatRules;
+
+    public function __construct()
+    {
+        $this->nonStrictFormatRules = new NonStrictFormatRules();
+    }
+
+    /**
+     * Alpha
+     *
+     * @param mixed|null $str
+     */
+    public function alpha($str = null): bool
+    {
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->alpha($str);
+    }
+
+    /**
+     * Alpha with spaces.
+     *
+     * @param string|null $value Value.
+     *
+     * @return bool True if alpha with spaces, else false.
+     */
+    public function alpha_space(?string $value = null): bool
+    {
+        if ($value === null) {
+            return true;
+        }
+
+        if (! is_string($value)) {
+            return false;
+        }
+
```

Comment:
With strict typing turned on, I think we'll not reach here as PHP will throw a `TypeError` already for passing not `null|string`. To reach here, then the typehint on the parameter should be removed.

Reply:
Thanks. All typehints for the value must be removed. Otherwise implicit type conversion or TypeError occurs.

Reply:
Fixed.